packages/text-input/text-input.scss

```diff
@@ -1,183 +1,188 @@
-        }
+        margin: rem(-2px) rem(4px);
     }
 
+    /* Modifications for input with action */
     &--action {
-        position: relative;
-
-        .jkl-text-field__input {
-            padding-right: $component-spacing--small + $icon-size;
+        .jkl-text-input__input {
+            padding-right: $text-input-height;
         }
     }
 
-    @include compact-mode {
-        & .jkl-text-field__input,
-        & .jkl-text-field__input::placeholder {
+    /* Modifications for compact variant (also inline) */
+    *[data-compactlayout] &,
+    &--compact,
+    &--inline {
+        .jkl-text-input__input {
             @include jkl-text-style("compact/body");
-        }
-        & .jkl-text-field__input {
-            height: $input-height--small-device;
             font-weight: $bold-weight;
+            height: $text-input-height--compact;
         }
-        & .jkl-text-input__action-button {
-            height: $input-height--small-device;
+        .jkl-text-input__action-button {
+            height: $text-input-height--compact;
         }
-        & .jkl-text-field__action-icon {
-            width: $icon-size--compact;
+        .jkl-text-input__action-icon {
+            width: rem(18px);
         }
     }
 
-    @include small-device {
-        .jkl-text-field__input,
-        .jkl-text-field__input::placeholder {
-            @include jkl-text-style("compact/body");
-        }
-        & .jkl-text-field__input {
-            height: $input-height--small-device;
-            font-weight: $bold-weight;
-        }
-        & .jkl-text-input__action-button {
-            height: $input-height--small-device;
-        }
-        & .jkl-text-field__action-icon {
-            width: $icon-size--compact;
+    /* Modifications for dark mode */
+    /* *[data-theme="dark"] &, */
+    &--inverted {
+        & .jkl-text-input__input {
+            color: $hvit;
+            background-color: $gra-90;
+            box-shadow: inset 0 rem(-1px) 0 0 $gra-50;
+
+            &::placeholder {
+                color: $gra-30;
+            }
+
+            &::selection {
+                background-color: $gra-60;
+            }
+
+            &:focus,
+            &:hover {
+                box-shadow: inset 0 rem(-2px) 0 0 $info--darkbg;
+            }
+
+            &[aria-invalid="true"],
+            &:invalid {
+                color: $feil--darkbg;
+                background-color: transparentize($feil--darkbg, 0.84);
+                box-shadow: inset 0 rem(-2px) 0 0 $feil--darkbg;
+
+                &::selection {
+                    background-color: transparentize($feil--darkbg, 0.75);
+                }
+            }
         }
     }
 }
 
+.jkl-text-area {
+    height: auto;
+    width: 100%;
+    & .jkl-text-input__input {
+        @include motion("standard");
+        transition-property: height, box-shadow;
+        font-weight: normal;
+        width: 100%;
+        resize: none; // Disable textarea resize
```

Comment:
https://catalin.red/css-resize-none-is-bad-for-ux/

Reply:
Ja, jeg er egentlig heller ingen fan av den. Ble vel satt inn fordi `transition` på `height` gjør at resizingen føles "seig". Men dette er en litt vel enkel løsning på det problemet 😅